src/jit/morph.cpp

```diff
@@ -4491,6 +4491,19 @@ void                Compiler::fgMoveOpsLeft(GenTreePtr tree)
             new_op1->gtType = TYP_I_IMPL;
         }
 
+        // If new_op1 is a new expression. Assign it a new unique value number.
+        if (new_op1->gtVNPair.GetLiberal() != ValueNumStore::NoVN)
+        {
+            noway_assert(ad1->gtVNPair.GetLiberal() != ValueNumStore::NoVN);
+            noway_assert(ad2->gtVNPair.GetLiberal() != ValueNumStore::NoVN);
+
+            // Since op is commutative, comparing only ad2 and op1 is enough.
+            if (ad2->gtVNPair.GetLiberal() != op1->gtVNPair.GetLiberal())
+            {
+                new_op1->gtVNPair.SetBoth(vnStore->VNForExpr(new_op1->TypeGet()));
```

Comment:
You could add this here:

  noway_assert(ad2->gtVNPair.GetLiberal() != ValueNumStore::NoVN);